src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -96,9 +96,23 @@ protected void ShutdownConnectedNodes(List<NodeContext> contextsToShutDown, bool
             // Send the build completion message to the nodes, causing them to shutdown or reset.
             _processesToIgnore.Clear();
 
+            // We wait for child nodes to exit to avoid them changing the terminal
+            // after this process terminates.
+            bool waitForExit =  !enableReuse &&
+                                !Console.IsInputRedirected &&
```

Comment:
When the streams are redirected, the application can not change terminal settings.

.NET uses stdin to know if it can change terminal settings:

https://github.com/dotnet/runtime/blob/2f1def8c556403ca85d1de50609492fef664b638/src/libraries/Native/Unix/System.Native/pal_console.c#L455